<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Preference</class>
 <widget class="QDialog" name="Preference">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>460</width>
    <height>546</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Preference</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <item>
    <widget class="QTabWidget" name="preferenceSwitch">
     <property name="styleSheet">
      <string notr="true">QTabWidget::pane {border-top:0px solid #e8f3f9;background:  transparent; }
QTabBar::tab {min-height: 100px;min-width:40px;color: white;border: 0px solid;border-top-left-radius: 0px;border-top-right-radius: 0px;padding:3px;}
QTabBar::tab:!selected {color: gray; margin-top: 5px; background-color: rgb(255,255,255,200)}
QTabBar::tab:selected {color: black; border: 1px solid; background-color: rgb(255,255,255,255)}</string>
     </property>
     <property name="tabPosition">
      <enum>QTabWidget::West</enum>
     </property>
     <property name="tabShape">
      <enum>QTabWidget::Rounded</enum>
     </property>
     <property name="currentIndex">
      <number>0</number>
     </property>
     <property name="elideMode">
      <enum>Qt::ElideNone</enum>
     </property>
     <property name="usesScrollButtons">
      <bool>true</bool>
     </property>
     <property name="documentMode">
      <bool>false</bool>
     </property>
     <property name="tabsClosable">
      <bool>false</bool>
     </property>
     <property name="movable">
      <bool>false</bool>
     </property>
     <property name="tabBarAutoHide">
      <bool>false</bool>
     </property>
     <widget class="QWidget" name="account">
      <attribute name="title">
       <string>&amp;Account</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="QGroupBox" name="groupBox_3">
         <property name="title">
          <string>name settings:</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_4">
          <item row="0" column="0">
           <widget class="QCheckBox" name="autoName_cb">
            <property name="text">
             <string>auto use your official name</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <layout class="QHBoxLayout" name="horizontalLayout_7">
            <item>
             <widget class="QLabel" name="playerName_label">
              <property name="text">
               <string>player name:</string>
              </property>
              <property name="buddy">
               <cstring>playerName_le</cstring>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="playerName_le"/>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="Line" name="line">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_4">
         <property name="title">
          <string>account settings:</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_6">
          <item row="0" column="0">
           <layout class="QHBoxLayout" name="horizontalLayout_8">
            <item>
             <widget class="QLabel" name="logName_label">
              <property name="text">
               <string>You have not logged in.</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="login_pb">
              <property name="text">
               <string>&amp;Log in</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>250</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="java">
      <attribute name="title">
       <string>&amp;Java</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QGroupBox" name="groupBox">
         <property name="title">
          <string>memory settings:</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_6">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <item>
             <widget class="QLabel" name="minMem_label">
              <property name="text">
               <string>minimum memory allocate:</string>
              </property>
              <property name="buddy">
               <cstring>minMem_sb</cstring>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="minMem_sb">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="maximum">
               <number>999999999</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout">
            <item>
             <widget class="QLabel" name="maxMem_label">
              <property name="text">
               <string>maximum memory allocate:</string>
              </property>
              <property name="buddy">
               <cstring>maxMem_sb</cstring>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="maxMem_sb">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="maximum">
               <number>999999999</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="Line" name="line_2">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_5">
         <property name="title">
          <string>JVM settings:</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_5">
          <item row="0" column="0">
           <layout class="QHBoxLayout" name="horizontalLayout_5">
            <item>
             <widget class="QLabel" name="javaPath_label">
              <property name="text">
               <string>Java path:</string>
              </property>
              <property name="buddy">
               <cstring>javaPath_le</cstring>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="javaPath_le"/>
            </item>
            <item>
             <widget class="QPushButton" name="autoJava_pb">
              <property name="text">
               <string>&amp;Auto</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="javaPath_showPb">
              <property name="text">
               <string>Show...</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="1" column="0">
           <layout class="QGridLayout" name="gridLayout_3">
            <item row="0" column="0">
             <widget class="QLabel" name="javaArg_label">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>extra Java arguement:</string>
              </property>
              <property name="buddy">
               <cstring>javaArg_te</cstring>
              </property>
             </widget>
            </item>
            <item row="0" column="1" rowspan="2">
             <widget class="QTextEdit" name="javaArg_te"/>
            </item>
            <item row="1" column="0">
             <spacer name="verticalSpacer">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>40</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_4">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>128</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="game">
      <attribute name="title">
       <string>&amp;Game</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <widget class="QGroupBox" name="groupBox_2">
         <property name="title">
          <string>path settings:</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="8" column="2">
           <widget class="QPushButton" name="assetsPath_showPb">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="text">
             <string>Show...</string>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QPushButton" name="more_pb">
            <property name="text">
             <string>More</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0" colspan="2">
           <widget class="QCheckBox" name="defaultPath_checkBox">
            <property name="text">
             <string>default relative path settings</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="corePath_le">
            <property name="enabled">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="versionsPath_label">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="text">
             <string>version folder path:</string>
            </property>
            <property name="buddy">
             <cstring>corePath_le</cstring>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="libsPath_label">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="text">
             <string>libraries path:</string>
            </property>
            <property name="buddy">
             <cstring>corePath_le</cstring>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="corePath_label">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="text">
             <string>core game path:</string>
            </property>
            <property name="buddy">
             <cstring>corePath_le</cstring>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QLineEdit" name="libsPath_le">
            <property name="enabled">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="6" column="1">
           <widget class="QLineEdit" name="nativesPath_le">
            <property name="enabled">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="6" column="2">
           <widget class="QPushButton" name="nativesPath_showPb">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="text">
             <string>Show...</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QLineEdit" name="versionsPath_le">
            <property name="enabled">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="2" column="2">
           <widget class="QPushButton" name="versionsPath_showPb">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="text">
             <string>Show...</string>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QPushButton" name="corePath_showPb">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="text">
             <string>Show...</string>
            </property>
           </widget>
          </item>
          <item row="10" column="2">
           <widget class="QPushButton" name="indexesPath_showPb">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="text">
             <string>Show...</string>
            </property>
           </widget>
          </item>
          <item row="12" column="0" colspan="3">
           <widget class="QTextBrowser" name="customPathHelper_tb">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Ignored">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>60</height>
             </size>
            </property>
            <property name="styleSheet">
             <string notr="true">background-color: rgba(255, 255, 255, 100);</string>
            </property>
            <property name="html">
             <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'SimSun'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:10pt;&quot;&gt;core game path:&amp;lt;core&amp;gt;&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:10pt;&quot;&gt;   assets path:&amp;lt;assets&amp;gt;&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:10pt;&quot;&gt;  laucher path:&amp;lt;launcher&amp;gt;&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
           </widget>
          </item>
          <item row="11" column="0">
           <widget class="QLabel" name="objectsPath_label">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="text">
             <string>asset objects path:</string>
            </property>
            <property name="buddy">
             <cstring>corePath_le</cstring>
            </property>
           </widget>
          </item>
          <item row="11" column="1">
           <widget class="QLineEdit" name="objectsPath_le">
            <property name="enabled">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="11" column="2">
           <widget class="QPushButton" name="objectsPath_showPb">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="text">
             <string>Show...</string>
            </property>
           </widget>
          </item>
          <item row="4" column="2">
           <widget class="QPushButton" name="libsPath_showPb">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="text">
             <string>Show...</string>
            </property>
           </widget>
          </item>
          <item row="10" column="1">
           <widget class="QLineEdit" name="indexesPath_le">
            <property name="enabled">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="10" column="0">
           <widget class="QLabel" name="indexesPath_label">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="text">
             <string>asset indexes path:</string>
            </property>
            <property name="buddy">
             <cstring>corePath_le</cstring>
            </property>
           </widget>
          </item>
          <item row="8" column="1">
           <widget class="QLineEdit" name="assetsPath_le">
            <property name="enabled">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="8" column="0">
           <widget class="QLabel" name="assetsPath_label">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="text">
             <string>core assets path:</string>
            </property>
            <property name="buddy">
             <cstring>corePath_le</cstring>
            </property>
           </widget>
          </item>
          <item row="6" column="0">
           <widget class="QLabel" name="nativesPath_label">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="text">
             <string>natives path:</string>
            </property>
            <property name="buddy">
             <cstring>corePath_le</cstring>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="Line" name="line_3">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_6">
         <property name="title">
          <string>display settings:</string>
         </property>
         <layout class="QGridLayout" name="gridLayout">
          <item row="1" column="1">
           <layout class="QHBoxLayout" name="horizontalLayout_6">
            <item>
             <widget class="QLabel" name="height_label">
              <property name="text">
               <string>height:</string>
              </property>
              <property name="buddy">
               <cstring>height_sb</cstring>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="height_sb">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="maximum">
               <number>999999999</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="1" column="0">
           <layout class="QHBoxLayout" name="horizontalLayout_3">
            <item>
             <widget class="QLabel" name="width_label">
              <property name="text">
               <string>width:</string>
              </property>
              <property name="buddy">
               <cstring>width_sb</cstring>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="width_sb">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="maximum">
               <number>999999999</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="0" column="0" colspan="2">
           <widget class="QCheckBox" name="fullScreen_checkBox">
            <property name="text">
             <string>full screen</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_3">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>193</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>corePath_le</tabstop>
  <tabstop>corePath_showPb</tabstop>
  <tabstop>width_sb</tabstop>
  <tabstop>height_sb</tabstop>
  <tabstop>maxMem_sb</tabstop>
  <tabstop>minMem_sb</tabstop>
  <tabstop>javaPath_le</tabstop>
  <tabstop>autoJava_pb</tabstop>
  <tabstop>javaPath_showPb</tabstop>
  <tabstop>javaArg_te</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>Preference</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>Preference</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
